### PR TITLE
Simplify naming in `i257`

### PR DESCRIPTION
**Pull request by delaaxe**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the new behavior?

While code like the following is eventually understandable, it isn't as clearly obvious as the proposed one:

Before:

```rust
fn foo(value: i257) {
  let mut x: u256 = 42;
  if value.sign == true {
    x += value.inner;
  }
}
```

After: 

```rust
fn foo(value: i257) {
  let mut x: u256 = 42;
  if value.is_negative {
    x += value.abs;
  }
}
```

## Does this introduce a breaking change?

- [X] Yes
- [ ] No

### CODE REVIEW

**Comment by LucasLvy** [comment]
@shramee  any opinion ?

**Comment by shramee** [comment]
Yeah, I think `is_negative` that makes it clearer! ❤️

File: src/math/src/signed_u256.cairo [hunk 1]
```diff
@@ -225,36 +124,192 @@ impl i257PartialOrd of PartialOrd<i257> {
...
+    }
+}
+
+// Adds two i257 integers.
+// # Arguments
+// * `a` - The first i257 to add.
+// * `b` - The second i257 to add.
+// # Returns
+// * `i257` - The sum of `a` and `b`.
+fn i257_add(lhs: i257, rhs: i257) -> i257 {
```
Comment (LucasLvy) [inline src/math/src/signed_u256.cairo:182]: why did you move all this functions outside of the traits ?

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: src/math/src/signed_u256.cairo (hunk 1 above)
Comment (delaaxe) [reply src/math/src/signed_u256.cairo:182]: to have a cleaner diff with `i128` which is also used in some projects

**Comment by delaaxe** [comment]
If you're ok I'd also do

```rust
struct i257 {
    abs: u256,
    is_negative: bool,
}
```

instead of 

```rust
struct i257 {
    is_negative: bool,
    abs: u256,
}
```

as it would make storage packing a bit more intuitive if ever needed

File: src/math/src/signed_u256.cairo (hunk 1 above)
Comment (delaaxe) [reply src/math/src/signed_u256.cairo:182]: or at least, the corelib implementation wasn't sufficient enough and that's why it needed to be redone in satoru iirc

Comment (LucasLvy) [reply src/math/src/signed_u256.cairo:182]: keep them in the traits plz

Comment (delaaxe) [reply src/math/src/signed_u256.cairo]: done

**Review by LucasLvy — APPROVED** [review]
